### DISCUSSION TITLE
Latest Genmon update / no email notification from exercise cycle.

### DISCUSSION THREAD

**Discussion by Jkh62** [jgyates/genmon#1105] [General]
Updated to the latest software from the email notice I received. Was out working in the garage and my Gen exercise cycle started but no email notification.

Double checked my time and date (correct) then set a new exercise time 30 minutes later.
Gen exercise cycled without any email.

Did I miss a setting after the update?

**Comment by Jkh62** [comment]
Just sent my log /register files.

**Comment by skipfire** [comment]
For whatever it's worth, my emails did fire after the update.

**Comment by Jkh62** [comment]
Did you reboot your PI?

**Comment by jgyates** [comment]
I received your logs. No errors that I can see.

The software will read modbus register 1 and 2 as a single modbus register. The high part is the engine state and the low part is the switch state and alarm state. If there is any change in this register, and email is sent. You registers that you submitted indicate that you have maintenance alarm pending. This makes the low part not change regardless of the switch state as long as the alarm condition is active. 

We can do this as a test: You can run your generator and see if you get emails. The emails sent during exercise are just from monitoring the state of registers 1 and 2.

Let me know how the test goes.

**Reply by Jkh62** [reply]
Run manually I get notification.

**Reply by jgyates** [reply]
I can't think of anything that has changed that may have caused this. Here are few things to check.

check your spam folder. 

Based on the registers it looks like the maintenance alarm may have occurred during during the exercice cycle. This could have effected this way the registers are returned during the exercise, but that is just a guess. You logs have multiple entries for the maintenance that overlap your exercise cycle:

        04/20/24 13:30:49 Service Schedule A
        04/20/24 12:46:17 Service Schedule A
        04/20/24 12:10:16 Service Schedule A

Here are the entries from the run log:

        04/20/24 13:30:49 Stopped - Auto
        04/20/24 13:25:37 Running - Exercise
        04/20/24 12:46:17 Stopped - Auto
        04/20/24 12:40:34 Running - Remote Start
        04/20/24 12:10:16 Stopped - Auto
        04/20/24 12:05:04 Running - Exercise

I would suggest that you set a reminder to check your next exercise cycle and see if you get emails, or change you exercise cycle temporarily and see if you can duplicate the issue. If you do the latter you may consider resetting your maintenance alarm as this may have an effect on the controllers status.

**Comment by Jkh62** [comment]
Just tested by changing exercise time and no notification.
I did reset the alarm status thru genmon. Is that what you referenced or do it at the generator controller ?

Exercise cycle is spot on for start and stop time , just no email sent.

Strange......

**Reply by skipfire** [reply]
Have you reviewed the Notifications tab to make sure that didn't change (or change it and change it back to be sure it resets)?  Since the logs were sent the email seems to be configured right, maybe the recipient list got lost?

**Reply by Jkh62** [reply]
Will check that.
Doing a pi system update & waiting for it to finish and reboot.

**Reply by jgyates** [reply]
I don't think resetting the alarms with the remote command will reset the maintenance alarm, but I am not 100 percent sure (only about 80 percent). You can tell if it is reset by looking at the web interface and seeing the alarm is active on the main status page. If the system is in alarm you will see these two messages on the Status page:

        Switch State : System in Alarm
        System In Alarm : Service A Due

If you still see these then you would need to reset the maintenance on the controller. The remote restet alarm does not reset all conditions and if memory serves I don't think it resets the maintenance alarm as to do this you have to explicitly reset the maintenance counter in the controller.

This is a video that shows how to reset the maitenace timer on Evo 2.0:  https://www.youtube.com/watch?v=bjyUUSa2-O4&ab_channel=BrendanKelley

Let me know how the test goes once you reset the maintenance alarm.

**Reply by jgyates** [reply]
@Jkh62 said he was receiving notices on a manual start and when genmon starts so that would use the same email on the notification page. 

My guess is that this is some issue with the Evo2.0 regarding exercising while a maintenance alarm is active.

Also, the code that send the email for starting is agnostic regarding if the generator is starting for exercise or manually, so this makes me think it is a controller anomaly.

**Comment by Jkh62** [comment]
....still waiting on PI system update to finish.

Went outside and had visible yellow alarm light , that was a surprise and maybe just really bad timing and appeared before the exercise cycle. Its reset now and will test.

**Comment by Jkh62** [comment]
Does the Pi system update wipe out / reset my Log2ram ?

**Reply by skipfire** [reply]
I don't believe an update should impact log2ram.

**Reply by jgyates** [reply]
No, log2ram will write to a RAM disk and flush it to disk periodically in block writes. Every write to a SD card must be done in blocks so saving the writes to do them all at once is better for the life of the SD card.

**Comment by Jkh62** [comment]
Back to normal / exercise update email notification working normal.

I did not notice an impending alarm or alarm status before doing the Genmon update. 
After going into the dealer section of the controller at the generator and resetting the visible yellow alarm status I am pretty sure that fixed the issue. I might have created it by doing a Genmon update during a pending then visible alarm status.

I did get my Pi OS updated which I was curious on how long it takes and it took a while but its not a bad thing to do.

Looks like this was entirely bad timing on my part.
If I have any luck its bad luck .

Thx for the help guys!

**Reply by jgyates** [reply]
I am glad it is working for you.

**Comment by Jkh62** [comment]
Again, Thx for the help.

I think schedule A maintenance is the battery check.
If it showed in the log file / register did it show on genmon on the web browser interface page?
Still learning the ropes on genmon and what to watch. Awesome software.

**Reply by jgyates** [reply]
yes. If there is any alarm condition the menu in the web interface turns yellow, except for when maintainence is due, then it turns yellow. Also, an email is sent on the alarm condition (e.g. when register 1 and 2 change in an alarm condition) and the text of the web interace shows an alarm state. Below is the screen cap of my simulator running. with your regiseters. Note the yellow menu (green is no alarm, ready to run).

<img width="601" alt="Screenshot 2024-04-20 at 5 15 29 PM" src="https://github.com/jgyates/genmon/assets/4513286/b66c5d10-43db-4e49-a677-0174893fa60a">

Also, the logs on the log page are maintained by the controller, genmon just displays them. The outage log and power log (power graph) are maintained by genmon.

**Comment by Jkh62** [comment]
That color indicator is genius and a nice touch , I learned a few new things today.......
Again , thank you VERY much for taking the time to look at my logs / register and explain everything.
